src/main/java/com/iexec/commons/poco/chain/AbstractAssetDeploymentService.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2024 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.commons.poco.chain;
+
+import com.iexec.commons.poco.encoding.AssetDataEncoder;
+import lombok.extern.slf4j.Slf4j;
+import org.web3j.utils.Numeric;
+
+import java.io.IOException;
+import java.math.BigInteger;
+
+@Slf4j
+public abstract class AbstractAssetDeploymentService {
+    protected final SignerService signerService;
+
+    private final String assetRegistrySelector;
+    private String assetRegistryAddress;
+
+    protected AbstractAssetDeploymentService(SignerService signerService, String assetRegistrySelector) {
+        this.signerService = signerService;
+        this.assetRegistrySelector = assetRegistrySelector;
+    }
+
+    public void initRegistryAddress(String iexecHubAddress) throws IOException {
+        if (this.assetRegistryAddress != null) {
+            return;
+        }
+        this.assetRegistryAddress = toEthereumAddress(signerService.sendCall(iexecHubAddress, assetRegistrySelector));
+    }
```

Comment:
It would be great to annotate this method with  `PostConstruct`. This way, it could be automatically called by the frameworks used for the dependent projects.

Reply:
Sadly, this cannot be done today as the blockchain is not always up when creating an instance.
It is something to keep in mind for the future.